### ISSUE TITLE
Transport SSH not working since 2.4.3.0

### ISSUE THREAD

**Issue by 0xd3e** [ansible/ansible#35730]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
Inventory / SSH connection

##### ANSIBLE VERSION
```
ansible 2.4.3.0
  config file = /Users/danieltrautmann/Development/Projects/<path/to/>/ansible.cfg
  configured module search path = [u'/Users/danieltrautmann/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.4.3.0/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### CONFIGURATION
```
[defaults]
transport = ssh

[privilege_escalation]
become = true
```

##### OS / ENVIRONMENT
macOS High Sierra (Version 10.13.3)
Installed via Homebrew

##### SUMMARY
After upgrading from version 2.4.2.0 to 2.4.3.0 I've realized that the connection to the hosts doesn't work anymore.

When setting the transport to `paramiko`, the connection works well, but than the synchronize module fails.

However, the main issue here is that Ansible doesn't work with ssh.
Connecting to the hosts manually works.

Ansible only tries "well known" SSH key files, but the key I use has a different name.

```
$ ansible -vvvvv -i production -m ping all | grep -o -E 'Trying private key: [^\\]+'
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
Trying private key: /Users/danieltrautmann/.ssh/id_rsa
Trying private key: /Users/danieltrautmann/.ssh/id_dsa
Trying private key: /Users/danieltrautmann/.ssh/id_ecdsa
Trying private key: /Users/danieltrautmann/.ssh/id_ed25519
```

##### STEPS TO REPRODUCE
- Set transport to `ssh` (should be default)
- ansible -i <inventory> -m ping all

##### EXPECTED RESULTS

```
<HOST> | SUCCESS => {
    "changed": false,
    "ping": "pong"
}
```

##### ACTUAL RESULTS

```
<HOST> | UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: <USER>@<HOST>: Permission denied (publickey).\r\n",
    "unreachable": true
}
```

**Comment by 0xd3e** [comment]
It seems that my guess is correct. I've created a link named `id_rsa` to the key file. Now the connection works with `transport = ssh`.

```bash
$ ln -s <KEY FILE> id_rsa
```

This is a possible workaround besides using paramiko.

**Comment by sivel** [comment]
The lines that read `Trying private key` come from ssh itself.  Ansible does not try to guess where your private key is located, that is handled either by explicitly listing your private key via `ansible_ssh_private_key_file` or on the CLI as `--private-key`.

To instruct ssh, and by extension Ansible, you would have to indicate the paths to that key in your `ssh_config` file.

Perhaps you had some other configuration in place that assisted you?

needs_info

**Comment by 0xd3e** [comment]
Okay. Now I remembered that I've set `IdentitiesOnly yes` for all hosts in my SSH config recently.

This was done because when using Ansible with Vagrant, the temporary key for the Vagrant machine is added to the ssh-agent and when there are more than 6 keys, the authentication fails because SSH only checks 6 keys.

I've dropped `IdentitiesOnly yes` from my ssh config and now it works.
Thanks for pointing me to the right direction.
